lib/active_record_doctor/detectors/missing_non_null_constraint.rb

```diff
@@ -66,10 +66,20 @@ def required_presence_validators(model)
         model.validators.select do |validator|
           validator.is_a?(ActiveRecord::Validations::PresenceValidator) &&
             !validator.options[:allow_nil] &&
-            !validator.options[:if] &&
-            !validator.options[:unless]
+            (required_with_if?(validator) || !conditional_validator?(validator))
         end
       end
+
+      def required_with_if?(validator)
```

Comment:
If I understand correctly the intent here is checking whether `validator` was added by Rails as part of defining a `belongs_to` association. In this case, I suggest renaming this method to make the name capture that intent, e.g. `rails_belongs_to_presence_validator?`.